artichoke-backend/src/module_registry.rs

```diff
@@ -71,7 +46,7 @@ impl ModuleRegistry for Artichoke {
         let rclass = unsafe { self.with_ffi_boundary(|mrb| rclass.resolve(mrb))? };
         if let Some(mut rclass) = rclass {
             let module = unsafe { sys::mrb_sys_module_value(rclass.as_mut()) };
-            let module = Value::from(module);
+            let module = Self::Value::from(module);
```

Comment:
I like using `Self::Value` in the function signatures to match the trait definition, but in the function body, I'd prefer if we use the concrete type instead of indirecting through the trait.

```suggestion
            let module = Value::from(module);
```